### PR TITLE
Feature/log subtitle info

### PR DESCRIPTION
**Pull request by georgiana-b**
We should also truncate the file before writing a new subtitle since we're only interested in the current one.

### CODE REVIEW

**Comment by leplatrem** [comment]
Excellent — I'm moved — this is the first external contribution!

I have just one concern: the location of the file and the permission to write to that folder.

Here the location of the log depends a bit from how the software is ran: it goes to the current folder, am I right?
What about writing the log along the subtitle file (e.g. `D:\movie.srt` → `D:\movie.srt.log`) ?
Maybe it does not fulfill your usecase, just tell me.

What happens if the user running the software has no permission to write to that folder? For example, under MacOSX and Linux, the users have no permission to write in the application folder. Or something a disk can be full (think of USB sticks for eg.).
I think you should add a safety check in case the file could not be opened.

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -587,6 +589,13 @@ void MainWindow::actionOpen()
     if (!fileName.isEmpty()) {
         m_lastFolder = QFileInfo(fileName).absoluteDir().absolutePath();
         openFile(fileName);
```
Comment (leplatrem) [inline src/mainwindow.cpp]: Please leave a blank line and add a comment about what you are about to do (start a log file)

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -587,6 +589,13 @@ void MainWindow::actionOpen()
...
+        logFile->open(QIODevice::WriteOnly | QIODevice::Truncate);
```
Comment (leplatrem) [inline src/mainwindow.cpp]: Check the return value: `bool success = logFile->open()`

File: src/mainwindow.cpp [hunk 3]
```diff
@@ -872,6 +881,12 @@ void MainWindow::subtitleChanged(QList<Subtitle*> p_currentSubtitles)
         QWidget* withFocus = qApp->focusWidget();
         ui->tableWidget->selectRow(subtitleRow);
         if (withFocus) withFocus->setFocus();  // restore
+
+        logFile->open(QIODevice::WriteOnly | QIODevice::Append);
```
Comment (leplatrem) [inline src/mainwindow.cpp]: Check the return value here too

**Comment by georgiana-b** [comment]
@leplatrem  I apologize for the PR. I actually intended to open it on [our fork](https://github.com/mitzaM/subtivals). I see what you mean there but this is a very rushed implementation that was made to work on a Linux we configured particularly for a festival we have. I would love to do what you're proposing but I'm a noob in QT and a I written C++ for a long time so I don't really know how. Feel free to use the code I wrote so far. I'm closing this.

**Comment by leplatrem** [comment]
Oh no please! I'd love to merge that if that makes sense for you! It's a pity if you have to maintain a fork yourself... As you want!
It looks like your Qt/C++ are very fine!

I saw that you also improved stuff like «scroll to top» on subtitle change, I'd love to merge it if you think it's better!

As a side node, since what you appear to be doing is related to generating a JSON log file when subtitles are changed: did you check the **live feature** ?
Instead of writing to a log file, every time the subtitle is changed it sends a piece of JSON to a websocket.  A client receives it and can process. In our example, the websocket client is a simple webpage with a few lines of Javascript that show the subtitles on a webpage...

**Comment by georgiana-b** [comment]
@leplatrem You're right about the live feature. I only noticed it when I repackaged our fork. It does indeed provide the functionality we were looking for but I somehow missed it. We will use it next year. Gracias!

**Pull request closed without merging**